src/components/MagicCodeInput.js

```diff
@@ -137,17 +166,35 @@ function MagicCodeInput(props) {
     }, [props.value, props.shouldSubmitOnComplete]);
 
     /**
-     * Callback for the onFocus event, updates the indexes
-     * of the currently focused input.
+     * Focuses on the input when it is pressed.
      *
      * @param {Object} event
      * @param {Number} index
      */
-    const onFocus = (event, index) => {
+    const onFocus = (event) => {
+        if (shouldFocusLast.current) {
+            setInput(TEXT_INPUT_EMPTY_STATE);
+            setFocusedIndex(lastFocusedIndex.current);
+            setEditIndex(lastFocusedIndex.current);
+        }
         event.preventDefault();
-        setInput('');
+    };
+
+    /**
+     * Callback for the onPress event, updates the indexes
+     * of the currently focused input.
+     *
+     * @param {Number} index
+     */
+    const onPress = (index) => {
+        shouldFocusLast.current = false;
+        if (!Browser.isMobileChrome() && !Browser.isMobileSafari()) {
+            inputRefs.current.focus();
+        }
```

Comment:
Can you add comments here as it still unclear

Reply:
@wojtus7 bump

Reply:
Added comment 🫡